### ISSUE TITLE
Biased spherical integration in NewMultiScattCS

### ISSUE THREAD

**Issue by Ralith** [sebh/UnrealEngineSkyAtmosphere#12]
The multi-scattering precompute shader numerically integrates incoming light using sample rays distributed uniformly along polar coordinates: 
https://github.com/sebh/UnrealEngineSkyAtmosphere/blob/f446f242ebd4bde512446a1a3212a5e333a0adb5/Resources/RenderSkyRayMarching.hlsl#L448-L470
Uniformly distributed polar coordinates are more concentrated around the poles (close to phi = 0 or PI), and sparse around the equator (phi = PI/2). Traditionally this is compensated for by weighting samples by a `sin(phi)` factor, but that doesn't seem to be in evidence. Does NewMultiScattCS have an unintended bias, or is this compensated for elsewhere? If so, where?

As always, thanks for publishing this excellent paper and the very helpful example code regardless.

**Comment by sebh** [comment]
Hi @Ralith ,

You are correct! I remember fixing that at some point for the paper but I guess I messed up when I created that repository, not sure what happened... It should be fixed now with https://github.com/sebh/UnrealEngineSkyAtmosphere/commit/a0229bc5a008c83ee5df8c5b50d9bb3b9f6583fe (I have not ran in depth checks, I do not have the time to do that now).

Unreal does not have that issue sine it uses a low sample count approximation and the high quality multiple scattering has a correct uniform distribution.

Thank you for reporting!
